web/js/MapUI/Components/Markers/Markers.js

```diff
@@ -0,0 +1,199 @@
+import { getActiveLayers, getMaxZoomLevelLayerCollection } from '../../../modules/layers/selectors';
+
+const Markers = (props) => {
+  const {
+    action,
+    activeLayers,
+    config,
+    coordinates,
+    isMobileDevice,
+    selectedMap,
+    selectedMapMarkers,
+    proj,
+    removeMarker,
+    setGeocodeResults,
+    ui,
+  } = props;
+
+  useEffect(() => {
+    switch (action.type) {
+      case 'LOCATION_SEARCH/REMOVE_MARKER': {
+        return removeCoordinatesMarker(action.coordinates);
+      }
+      case 'LOCATION_SEARCH/SET_MARKER': {
+        if (action.flyToExistingMarker) {
+          return flyToMarker(action.coordinates);
+        }
+        return addMarkerAndUpdateStore(true, action.reverseGeocodeResults, action.isCoordinatesSearchActive, action.coordinates);
+      }
+      case 'LOCATION_SEARCH/TOGGLE_DIALOG_VISIBLE': {
+        return addMarkerAndUpdateStore(false);
+      }
+      default:
+        break;
+    }
+  }, [action]);
+
+  /*
+   * Remove coordinates marker from all projections
+   *
+   * @method removeCoordinatesMarker
+   * @static
+   *
+   * @returns {void}
+   */
+  const removeCoordinatesMarker = (coordinatesObject) => {
+    selectedMapMarkers.forEach((marker) => {
+      if (marker.id === coordinatesObject.id) {
+        marker.setMap(null);
+        selectedMap.removeOverlay(marker);
+      }
+    });
+  };
+
+  /**
+   * Remove all coordinates markers
+   *
+   * @method removeAllCoordinatesMarkers
+   * @static
+   *
+   * @returns {void}
+   */
+  const removeAllCoordinatesMarkers = () => {
+    ui.markers.forEach((marker) => {
+      marker.setMap(null);
+      ui.selected.removeOverlay(marker);
+    });
+  };
+
+  /**
+   * Handle reverse geocode and add map marker with results
+   *
+   * @method handleActiveMapMarker
+   * @static
+   *
+   * @returns {void}
+   */
+  const handleActiveMapMarker = () => {
+    removeAllCoordinatesMarkers();
+    if (coordinates && coordinates.length > 0) {
+      coordinates.forEach((coordinatesObject) => {
+        const { longitude, latitude } = coordinatesObject;
+        const coord = [longitude, latitude];
+        if (!areCoordinatesWithinExtent(proj, coord)) return;
+        reverseGeocode(getNormalizedCoordinate(coord), config).then((results) => {
+          addMarkerAndUpdateStore(true, results, null, coordinatesObject);
+        });
+      });
+    }
+  };
+
+  const flyToMarker = (coordinatesObject) => {
+    const { sources } = config;
+    const { longitude, latitude } = coordinatesObject;
+    const latestCoordinates = coordinatesObject && [longitude, latitude];
+    const zoom = selectedMap.getView().getZoom();
+    const maxZoom = getMaxZoomLevelLayerCollection(activeLayers, zoom, proj.id, sources);
+    animateCoordinates(selectedMap, proj, latestCoordinates, maxZoom);
+  };
+
+  /**
+   * Add map coordinate marker and update store
+   *
+   * @method addMarkerAndUpdateStore
+   * @static
+   *
+   * @param {Object} geocodeResults
```

Comment:
Add an entry for each parameter (currently only 2 of 4 are listed)

Reply:
Added param tags for remaining parameters.